### PR TITLE
Add DFN-10 package for pitch 0.5mm and EP 1.2x2

### PR DESCRIPTION
**Pull request by matthuszagh**

### CODE REVIEW

File: scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml [hunk 1]
```diff
@@ -253,6 +253,46 @@ DFN-8-1EP_3x3mm_P0.65mm_EP1.7x2.05mm:
...
+DFN-10-1EP_2.5x2.5mm_P0.5mm_EP1.2x2.0mm:
+  device_type: 'DFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'http://www.ti.com/lit/ds/symlink/tps7a91.pdf (page 30)'
+  # ipc_class: 'qfn' # 'qfn_pull_back'
+  #ipc_density: 'least' #overwrite global value for this device.
+  body_size_x:
+    nominal: 2.5
+    tolerance: 0.1
```
Comment (poeschlr) [inline scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml]:
```yaml
  body_size_x:
    minimum: 2.4
    maximum: 2.6
```

only minimum and maximum are given in the datasheet.

File: scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml [hunk 2]
```diff
@@ -253,6 +253,46 @@ DFN-8-1EP_3x3mm_P0.65mm_EP1.7x2.05mm:
...
+  body_size_y:
+    nominal: 2.5
+    tolerance: 0.1
+  lead_width:
+    nominal: 0.25
+    tolerance: 0.05
```
Comment (poeschlr) [inline scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml]:
```yaml
  lead_width:
    minimum: 0.2
    maximum: 0.3
```

File: scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml [hunk 3]
```diff
@@ -253,6 +253,46 @@ DFN-8-1EP_3x3mm_P0.65mm_EP1.7x2.05mm:
...
+  lead_len:
+    nominal: 0.4
+    tolerance: 0.05
```
Comment (poeschlr) [inline scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml]:
```yaml
  lead_len:
    minimum: 0.35
    maximum: 0.45
```

File: scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml [hunk 4]
```diff
@@ -253,6 +253,46 @@ DFN-8-1EP_3x3mm_P0.65mm_EP1.7x2.05mm:
...
+  EP_size_x:
+    nominal: 1.2
+    tolerance: 0.1
+  EP_size_y:
+    nominal: 2.0
+    tolerance: 0.1
+  # EP_paste_coverage: 0.65
+  EP_num_paste_pads: [2, 2]
+  # heel_reduction: 0.1 #for relatively large EP pads (increase clearance)
+
```
Comment (poeschlr) [inline scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml]: This one would need a thermal via version as well.
```yaml
thermal_vias:
    count: [2, 3]
    drill: 0.2
    # min_annular_ring: 0.15
    paste_via_clearance: 0.1
    EP_num_paste_pads: [2, 2]
    #paste_between_vias: 1
    #paste_rings_outside: 1
    EP_paste_coverage: 0.7
    #grid: [1, 1]
    # bottom_pad_size:
    paste_avoid_via: False
```

File: scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml [hunk 5]
```diff
@@ -253,6 +253,46 @@ DFN-8-1EP_3x3mm_P0.65mm_EP1.7x2.05mm:
...
+  #part_number: 'mpn'
+  size_source: 'http://www.ti.com/lit/ds/symlink/tps7a91.pdf (page 30)'
+  # ipc_class: 'qfn' # 'qfn_pull_back'
+  #ipc_density: 'least' #overwrite global value for this device.
+  body_size_x:
+    nominal: 2.5
+    tolerance: 0.1
+  body_size_y:
+    nominal: 2.5
+    tolerance: 0.1
```
Comment (poeschlr) [inline scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml]: please add the body height as well.

```yaml
body_height:
    minimum: 0.7
    maximum: 0.8
```

File: scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml [hunk 6]
```diff
@@ -253,6 +253,46 @@ DFN-8-1EP_3x3mm_P0.65mm_EP1.7x2.05mm:
   #include_suffix_in_3dpath: 'False'
   #include_pad_size: 'fp_name_only' # 'both' | 'none' (default)
 
+DFN-10-1EP_2.5x2.5mm_P0.5mm_EP1.2x2.0mm:
+  device_type: 'DFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'http://www.ti.com/lit/ds/symlink/tps7a91.pdf (page 30)'
```
Comment (poeschlr) [inline scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml]: This style of defining the page allows the web browser to open the datasheet at the correct page.

```yaml
  size_source: 'http://www.ti.com/lit/ds/symlink/tps7a91.pdf#page=30'
```

File: scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml [hunk 7]
```diff
@@ -253,6 +253,46 @@ DFN-8-1EP_3x3mm_P0.65mm_EP1.7x2.05mm:
...
+  # ipc_class: 'qfn' # 'qfn_pull_back'
+  #ipc_density: 'least' #overwrite global value for this device.
```
Comment (poeschlr) [inline scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml]: We might want to use the texas given package name for this footprint:

```yaml
custom_name_format: 'Texas_S-PDSO-N{pincount}_EP{ep_size_x:g}x{ep_size_y:g}mm{vias:s}'
```

File: scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml [hunk 8]
```diff
@@ -253,6 +253,46 @@ DFN-8-1EP_3x3mm_P0.65mm_EP1.7x2.05mm:
   #include_suffix_in_3dpath: 'False'
   #include_pad_size: 'fp_name_only' # 'both' | 'none' (default)
 
+DFN-10-1EP_2.5x2.5mm_P0.5mm_EP1.2x2.0mm:
```
Comment (poeschlr) [inline scripts/Packages/Package_DFN_QFN/size_definitions/dfn.yaml]: If we choose to use the texas naming convention for the footprint we might as well call the size definition entry by the same name.

```yaml
Texas_S-PDSO-N10_EP1.2x2.0mm
```

**Comment by matthuszagh** [comment]
Made the changes. Let me know if anything else.

**Comment by poeschlr** [comment]
@pointhi footprint is merged, this is good to go.

**Pull request merged**